parastell/invessel_build.py

```diff
@@ -256,6 +308,144 @@ def generate_components(self):
             self.Components[name] = component
             interior_surface = outer_surface
 
+    def _connect_ribs_with_tris_moab(self, rib1, rib2, reverse=False):
+        """Creat MBTRI elements add add them to a surface between two ribs.
+
+        Arguments:
+            rib1 (Rib object): First of two ribs to be connected.
+            rib2 (Rib object): Second of two ribs to be connected.
+            reverse (bool): Optional. Whether to reverse the connectivity of
+                the MBTRIs being generated. Defaults to False.
+
+        Returns:
+            mb_tris (list of Entity Handle): List of the entity handles of the
+                MBTRIs connecting the two ribs.
+        """
+        mb_tris = []
+        for rib_loci_index, _ in enumerate(rib1.rib_loci[0:-1]):
+            corner1 = rib1.mb_verts[rib_loci_index]
+            corner2 = rib1.mb_verts[rib_loci_index + 1]
+            corner3 = rib2.mb_verts[rib_loci_index + 1]
+            corner4 = rib2.mb_verts[rib_loci_index]
+            corners = [corner1, corner2, corner3, corner4]
+            mb_tris += create_moab_tris_from_verts(
+                corners, self.mbc, reverse=reverse
+            )
+        return mb_tris
+
+    def _generate_pymoab_verts(self):
+        """Generate MBVERTEX entities from rib loci in all surfaces"""
+        [
+            surface._generate_pymoab_verts(self.mbc)
+            for surface in self.Surfaces.values()
+        ]
+
+    def _generate_curved_surfaces_pydagmc(self):
+        """Generate the faceted representation of each curved surface and
+        add it to the PyDAGMC model, remembering the surface ids."""
+        self.curved_surface_ids = []
+        for surf_idx, surface in enumerate(self.Surfaces.values()):
+            mb_tris = []
+            for rib, next_rib in zip(surface.Ribs[0:-1], surface.Ribs[1:]):
+                mb_tris += self._connect_ribs_with_tris_moab(
+                    rib, next_rib, reverse=True if surf_idx == 0 else False
```

Comment:
```suggestion
                    rib, next_rib, reverse=(surf_idx == 0)
```

Reply:
Note: you could do this without `surf_idx` is you made this condition `surface == self.Surfaces.values()[0]`.

To make it more readable, you could define:
```
surfaces = self.Surfaces.values()
first_surface = surfaces[0]
```
and then loop over `surfaces` without enumerate, and use the condition `surface == first_surface`